compiler/aarch64/codegen/GenerateInstructions.cpp

```diff
@@ -44,30 +44,14 @@ TR::Instruction *generateInstruction(TR::CodeGenerator *cg, TR::InstOpCode::Mnem
    return new (cg->trHeapMemory()) TR::Instruction(op, node, cg);
    }
 
-TR::Instruction *generateImmInstruction(TR::CodeGenerator *cg, TR::InstOpCode::Mnemonic op, TR::Node *node, uint32_t imm,
+TR::ARM64ImmInstruction *generateImmInstruction(TR::CodeGenerator *cg, TR::InstOpCode::Mnemonic op, TR::Node *node, uint32_t imm,
```

Comment:
Should we also change the return types of other `generateXXXInstruction()` to make them consistent?

Reply:
Yes, but that's beyond the scope of this PR.  I created issue #4179 to track this work.